test/callbacks_test.rb

```diff
@@ -0,0 +1,200 @@
+require "test_helper"
+
+def with_mocked_record
+  person_mock = Minitest::Mock.new
+
+  Person.stub :new, person_mock do
+    person = Person.new
+    yield person_mock, person
+
+    assert_mock person_mock
+  end
+end
+
+def with_mocked_proc
+  proc_mock = Minitest::Mock.new
+
+  yield proc_mock
+
+  assert_mock proc_mock
+end
```

Comment:
Think we can get here without mocking? Would rather test the direct API. So maybe just setup a separate Person fixture model that has callbacks defined. Or even just defined them on the Person model, where all they do is set something you can check for.